internal/util/importutilv2/binlog/column_reader.go

```diff
@@ -0,0 +1,60 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package binlog
+
+import (
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+	"github.com/milvus-io/milvus/internal/storage"
+)
+
+type columnReader struct {
+	reader      *storage.BinlogReader
+	fieldSchema *schemapb.FieldSchema
+}
+
+func newColumnReader(cm storage.ChunkManager, fieldSchema *schemapb.FieldSchema, path string) (*columnReader, error) {
+	reader, err := newBinlogReader(cm, path)
+	if err != nil {
+		return nil, err
+	}
+	return &columnReader{
+		reader:      reader,
+		fieldSchema: fieldSchema,
+	}, nil
+}
+
+func (r *columnReader) Next(_ int64) (storage.FieldData, error) {
```

Comment:
It is important to note that if there is no data in "insertData" after filtering, returning an "insertData" with 0 rows may be mistakenly interpreted by the upper layer as all data being read.